framework/base/Controller.php

```diff
@@ -186,12 +213,12 @@ public function run($route, $params = [])
     {
         $pos = strpos($route, '/');
         if ($pos === false) {
-            return $this->runAction($route, $params);
+            return $this->runAction(Yii::$app->getRequest(), $route, $params);
```

Comment:
We had a discussion about injecting Request instance in controller. Seems to be useful in this case

Reply:
The only way to do this in the middleware scope is having a setter for the request instance allowing request to be re-set per each action execution in the same way `Controller::$action` behaves.
Such approach was rejected at #14645.
This matter should be solved afterwards.